monix-tail/shared/src/main/scala/monix/tail/internal/IterantTakeEveryNth.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2014-2018 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.tail.internal
+
+import monix.tail.Iterant.{Halt, Last, Next, NextBatch, NextCursor, Suspend}
+import monix.tail.Iterant
+import cats.syntax.all._
+import cats.effect.Sync
+import monix.execution.misc.NonFatal
+import monix.tail.batches.BatchCursor
+import monix.tail.internal.IterantUtils._
+
+import scala.collection.mutable.ArrayBuffer
+
+private[tail] object IterantTakeEveryNth {
+  /**
+    * Implementation for `Iterant#takeEveryNth`
+    */
+  def apply[F[_], A](source: Iterant[F, A], n: Int)
+                    (implicit F: Sync[F]): Iterant[F, A] = {
+
+    require(n > 0, "n must be strictly positive")
+
+    if (n == 1) return source
```

Comment:
I don't like usage of `return`, unless it's made for concrete optimization reasons, or in other words it's very low level stuff.

This check should be included before the `source match` below.

Reply:
@alexandru Totally agree. Just could not find a better place for it. Fixed.